Makefile

```diff
@@ -15,7 +15,7 @@ MAN = $(DESTDIR)/usr/share/man/man8
 INITD = $(DESTDIR)/etc/rc.d/init.d
 LIBDIR = $(DESTDIR)/usr/lib/dkms
 BASHDIR = $(DESTDIR)/usr/share/bash-completion/completions
-KCONF = $(DESTDIR)/etc/kernel
+KCONF = $(DESTDIR)/usr/lib/kernel
```

Comment:
This will cause all 4 scripts to be installed in /usr/lib/kernel which might cause problems:

 - the `kernel_install.d_dkms` has hardcoded `/etc/kernel` references
   - can you please add an extra commit  to this PR to sed `sed s/@FOO@/$KCONF/` the path?
 - Debian has hardcoded lookup for `/etc` and no `/usr` equivalent and bypasses `kernel-install`
   - feel free to follow-up with Debian people with an issue, path, other
 
IMHO having these in under `/usr/lib/kernel` would be great, we cannot have the above line just yet. So let's drop it.

Reply:
> the kernel_install.d_dkms has hardcoded /etc/kernel references 

Oops, my bad, I hadn't noticed that the path was hardcoded.

> can you please add an extra commit to this PR to sed sed s/@FOO@/$KCONF/ the path?

I think we can do even better! I have added a commit that makes the path of the postinst and prerm scripts that we call from dkms.install relative. This way they will always work regardless of what we set KCONF to, without resorting to sed.

> IMHO having these in under /usr/lib/kernel would be great, we cannot have the above line just yet. So let's drop it.

I have dropped this part of the patch. Users and packagers can always override it with `make install KCONF='$(DESTDIR)/usr/lib/kernel'` anyway.

Reply:
The runtime expansion is a no-go IMHO for a couple of reasons:
 - tools running as root avoid use full paths, unless it's unavoidable
 - as-is this will break Fedora because they copy the file to `/usr/` while the rest are in `/etc` - yes they can workaround that, but they should not have to

Please use my original suggestion - the Makefile has some examples.

Thank you

Reply:
> as-is this will break Fedora because they copy the file to /usr/ while the rest are in /etc - yes they can workaround that, but they should not have to

The same would happen if we use sed to adjust the path.

Also it would be trivial for Fedora to adjust the spec file to accommodate for this change. Simply remove the custom install line and override KCONF instead.

Reply:
> The same would happen if we use sed to adjust the path.

I cannot see how that would happen. Can you elaborate?

> I have not heard of this, and besides /etc/kernel/install.d is also a full path....

Yes `/etc/kernel/install.d` is full path. The proposed change (snippet below) isn't.

```bash
SCRIPT=$(readlink -f "$0")
SCRIPTPATH=$(dirname "$SCRIPT")

"$SCRIPTPATH/../postinst.d/dkms" "$2"
```

Reply:
> I cannot see how that would happen. Can you elaborate?

If we sed the path and then move the script, the path is still wrong.

In any case, your sed solution is not an option because it completely breaks relocatability of the DESTDIR. If we set DESTDIR to some temporary directory, then this directory would end up in the script. Gentoo at least does this since it is required if the package manager is to keep track of what is actually being installed. I strongly suspect other package management systems also rely on relocatability (i.e. make an install image somewhere, and merge this install image to the file system. As opposed to installing to the file system directly). If we start sed'ing DESTDIR into the installed files something is sure to break somewhere.

I would strongly suggest to use my relative path solution instead. In Gentoo we always aim to use relative paths, precisely because we rely on the relocatability of the install image.

Also I fundamentally disagree that when fixing some bug it is possible or even desirable to ensure backwards compatibility with workarounds for that very same bug. It happens all the time that I implement some downstream workaround before attempting to properly fix something upstream. Once the solution is integrated upstream, then I have to remove my downstream workaround. This is a very normal process, installation scripts (ebuild, rpm spec or otherwise) simply have to be adjusted every now and then.

Reply:
Another option is to simply integrate the postinst and prerm scripts into the main script. WDYT?

Reply:
> If we sed the path and then move the script, the path is still wrong.

That would be an issue if only DESTDIR is embedded.

> Another option is to simply integrate the postinst and prerm scripts into the main script. WDYT?

This will intentionally break Debian and likely other distros.

Sounds like it's better if we drop the path handling in this PR. It's been on my TODO list anyway.

Reply:
> > If we sed the path and then move the script, the path is still wrong.
> 
> That would be an issue if only DESTDIR is embedded.

Then I don't understand what you want to sed in. I thought you wanted to sed in KCONF, but this contains DESTDIR?

Reply:
Yes, I've naively forgotten than the dkms variables are borked and contain DESTDIR. Sorry for taking you on this wild ride.

The 40-dkms.install commit is fine.

I am looking at fixing the paths... unless my brain melts before hand.

Reply:
Aaaah now I understand.

I've dropped the second commit, leaving only the renaming of the plugin.

We can look into adjusting the install paths as @anbe42 suggested later. If we do it that way then a sed won't break the relocatability.